photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala

```diff
@@ -198,19 +202,22 @@ object PalDBIndexMap {
 
     private def fetch(): Unit = {
       if (_iter == null) {
-        val merged = indexMap._storeReaders.foldLeft(List[JMap.Entry[Any, Any]]().toIterator){ (l, r) =>
-          l ++ r.iterable().iterator().asScala
-        }
+        val merged = indexMap
+          ._storeReaders
+          .foldLeft(List[JMap.Entry[Any, Any]]().toIterator){ (l, r) =>
+            l ++ r.iterable().iterator().asScala
+          }
         _iter = merged
       }
 
       while (_iter.hasNext && _currentItem == null) {
         val entry = _iter.next()
-        if (entry.getKey().isInstanceOf[String]) {
-          val key = entry.getKey().asInstanceOf[String]
-          val i = indexMap.getPartitionId(key)
-          val value = entry.getValue().asInstanceOf[Int] + indexMap._offsets(i)
-          _currentItem = (key, value)
+        entry.getKey match {
```

Comment:
Nice! We could also match `entry.getValue` here and eliminate the other `asInstanceOf`, e.g.:

```scala
(entry.getKey, entry.getValue) match {
  case (key: String, entryValue: Int) =>
     val i = indexMap.getPartitionId(key)
     val value = entryValue + indexMap._offsets(i)
     _currentItem = (key, value)
  case _ =>
}
```